dbms/src/Storages/Transaction/ProxyFFI.cpp

```diff
@@ -161,51 +161,107 @@ uint8_t TryFlushData(EngineStoreServerWrap * server, uint64_t region_id, uint8_t
 
 RawCppPtr CreateWriteBatch(const EngineStoreServerWrap * dummy)
 {
-    UNUSED(dummy);
-    return GenRawCppPtr(new UniversalWriteBatch(), RawCppPtrTypeImpl::WriteBatch);
+    try
```

Comment:
I don't think we should do try catch in these methods. As we didn't always do try catch in every ffi method, `GenCppRawString`, `GcRawCppPtr` and `SetStore` are all examples.
I think the principle is that we should only try catch the ffi which involves IO operations. But these ffis only involves memory operation and are really simple which shouldn't be try-catched.

Reply:
I agree that `CreateWriteBatch` not be protect with try catch. However this is because the function only includes operations that rarely raises exceptions.
I think if a operation may raise exceptions, it should be protected by try-catch on the language boundary. Otherwise we can not find an easy way to examine why the program aborted abruptly.

Reply:
Yes, that's why I try catch in methods like `HandleConsumeWriteBatch` and `HandleReadPage`. But these methods just involves operations like allocate memory, push into a vector, clear a vector and other simple ops.